content/sensu-go/5.17/reference/secret-providers.md

```diff
@@ -0,0 +1,297 @@
+required     | true
+type         | String
+example      | {{< highlight shell >}}"token": "VAULT_TOKEN"{{< /highlight >}}
+
+version      | 
+-------------|------ 
+description  | HashiCorp Vault [HTTP API version][14].
+required     | true
+type         | String
+example      | {{< highlight shell >}}"version": "v1"{{< /highlight >}}
+
+tls          | 
+-------------|------ 
+description  | TLS object. Vault only works with TLS configured. You may need to set up a CA cert if it is not already stored in your operating system's trust store. To do this, set the TLS object, and provide the `ca_cert` path. You may also need to set up a client cert, client key, or cname.
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"tls": {
+  "ca_cert": "/etc/ssl/certs/vault_ca_cert.pem",
+  "client_cert": "/etc/ssl/certs/vault_cert.pem",
+  "client_key": "/etc/ssl/certs/vault_key.pem",
+  "cname": "vault_client.example.com"
+}
+{{< /highlight >}}
+
+#### Rate limiter attributes
+
+limit        | 
+-------------|------ 
+description  | Maximum number of secrets requests per second that can be transmitted to the backend with the agent secrets API.
+required     | false
+type         | Float
+example      | {{< highlight shell >}}"limit": 10.0{{< /highlight >}}
+
+burst        | 
+-------------|------ 
+description  | Maximum amount of burst allowed in a rate interval for the secrets API.
+required     | false
+type         | integer
+example      | {{< highlight shell >}}"burst": 100{{< /highlight >}}
+
+## Secrets providers examples
+
+### VaultProvider example
+
+The `VaultProvider` secrets provider is a vendor-specific implementation for [HashiCorp Vault][5] secrets management.
+
+{{< language-toggle >}}
+
+{{< highlight yml >}}
+---
+type: VaultProvider
+api_version: secrets/v1
+metadata:
+  name: vault1
+spec:
+  client:
+    address: https://vaultserver.example.com:8200
+    token: VAULT_TOKEN
+    version: v1
+    tls:
+      ca_cert: "/etc/ssl/certs/vault_ca_cert.pem"
+    max_retries: 2
+    timeout: 20s
+    rate_limiter:
+      limit: 10
+      burst: 100
+{{< /highlight >}}
+
+{{< highlight json >}}
+{
+  "type": "VaultProvider",
+  "api_version": "secrets/v1",
+  "metadata": {
+    "name": "vault1"
+  },
+  "spec": {
+    "client": {
+      "address": "https://vaultserver.example.com:8200",
+      "token": "VAULT_TOKEN",
+      "version": "v1",
+      "tls": {
+        "ca_cert": "/etc/ssl/certs/vault_ca_cert.pem"
+      },
+      "max_retries": 2,
+      "timeout": "20s",
+      "rate_limiter": {
+        "limit": 10.0,
+        "burst": 100
+      }
+    }
+  }
+}
+{{< /highlight >}}
+
+{{< /language-toggle >}}
+
+### Env example
+
+Sensu's built-in `Env` secrets provider exposes secrets from backend [environment variables][4].
+Using the `Env` secrets provider may require you to synchronize environment variables in Sensu backend clusters.
+
+{{< language-toggle >}}
```

Comment:
This very resource definition is very minimal, which makes sense for the Env provider, but it could lead to some confusion. Consider add a note explaining that the Env provider isn't configurable outside of metadata.name, but does need to be explicitly enabled by providing a config like this one (if that is in fact the case -- I'm just guessing).

Reply:
I am not sure either. I will follow up with Eric about this. Eric and I talked about creating a docs issue to add a guide for using Env-based secrets management too.

Reply:
I added Env to the secrets management guide in https://github.com/sensu/sensu-docs/pull/2130